survivalstan/formulas.py

```diff
@@ -0,0 +1,315 @@
+        else:
+            group_id = None
+
+        if self._type == 'long':
+            return(self._prep_long(timepoint_id=timepoint_id, event_status=event_status,
+                                  subject_id=subject_id, group_id=group_id,
+                                  meta_data=meta_data, stan_data=stan_data)
+                  )
+        elif self._type == 'wide':
+            return(self._prep_wide(time=time, event_status=event_status, group_id=group_id,
+                                  meta_data=meta_data, stan_data=stan_data))
+
+surv = patsy.stateful_transform(Surv)
+
+def _get_args(s):
+    ''' Given a string of named code, return dict of named arguments
+    '''
+    pattern = r'(\w[\w\d_]*)\((.*)\)$'
+    match = re.match(pattern, s)
+    if match and len(match.groups())==2:
+        return dict(re.findall(r'(\S+)=(".*?"|[^ ,]+)', match.groups()[1]))
+    elif match is not None:
+        return list(match.groups())
+    else:
+        return []
+
+class SurvivalFactor(patsy.EvalFactor):
+    ''' A factor object to encode LHS variables
+        for Survival Models, including model type
+    '''
+    def __init__(self, *args, **kwargs):
+        super(SurvivalFactor, self).__init__(*args, **kwargs)
+        self._is_survival = False
+        self._class = None
+
+    def eval(self, *args, **kwargs):
+        result = super(SurvivalFactor, self).eval(*args, **kwargs)
+        try:
+            self._class = result.__class__
+        except:
+            logger.warning('Outcome class could not be determined')
+        if isinstance(result, SurvData):
+            self.code_args = _get_args(self.code)
+            self._is_survival = True
+            self._type = result.survival_type
+            self._meta_data = result.meta_data
+            self._stan_data = result.stan_data
+
+        return result
+
+class SurvivalModelDesc(object):
+    ''' A ModelDesc class to force use of SurvivalFactor when encoding LHS
+        variables for a SurvivalModel
+
+        Example:
+            
+            # simple survival model
+            my_formula = SurvivalModelDesc('surv(time=time,
+                event_status=event_value) ~ X1')
+            y, X = patsy.dmatrices(my_formula, data=df)
+
+            # with a subject id
+            my_formula2 = SurvivalModelDesc('surv(time=time,
+                    event_status=event_value, subject=subject_id) ~ X1')
+            y2, X2 = patsy.dmatrices(my_formula2, data=df)
+
+            # saves information about class & type of survival model
+            y2.design_info.terms[0].factors[0]._class
+            y2.design_info.terms[0].factors[0]._type
+    '''
+    def __init__(self, formula):
+        self.formula = formula
+        try:
+            self.lhs, self.rhs = re.split(string=formula, pattern='~', maxsplit=1)
+        except ValueError:
+            self.rhs = formula
+            self.lhs = ''
+        self.lhs_termlist = [patsy.Term([SurvivalFactor(self.lhs)])]
```

Comment:
Yep - this is the main value-add of using using SurvivalModelDesc class, to force the use of `SurvivalFactor` for lhs. I will likely also end up adding a new factor-class for rhs termlist (per issue #47, for example) but I haven't gotten there yet.